### PR TITLE
GUACAMOLE-1976: Add documentation for the OPTIONAL token modifier.

### PR DESCRIPTION
**Pull request by necouchman**

### CODE REVIEW

File: src/configuring-guacamole.md [hunk 1]
```diff
@@ -2755,6 +2755,21 @@ supported:
...
+: Typically when tokens are specified as part of a conneciton parameter,
+  if the token is undefined, the literal value of the token will be placed
+  in the connection parameter. This modifier marks a token as "optional",
+  which means that an undefined token will not be added as a literal and
+  will simply be skipped as if the toekn is not entered.
+
+  In some cases this will simply result in a string lacking the information
+  that would otherwise be in the token; however, in other cases this may
+  result in Guacamole generating prompts. For example, if you specify the
+  oken `${GUAC_PASSWORD}` for the password field in a connection, but you
```
Comment (jmuehlner) [inline src/configuring-guacamole.md:2768]: Typo: "oken"

File: src/configuring-guacamole.md [hunk 2]
```diff
@@ -2755,6 +2755,21 @@ supported:
 `UPPER`
 : Convert the entire value of the token to upper-case.
 
+`OPTIONAL`
+: Typically when tokens are specified as part of a conneciton parameter,
```
Comment (jmuehlner) [inline src/configuring-guacamole.md:2759]: Typo: "conneciton"

File: src/configuring-guacamole.md [hunk 3]
```diff
@@ -2755,6 +2755,21 @@ supported:
...
+  if the token is undefined, the literal value of the token will be placed
+  in the connection parameter. This modifier marks a token as "optional",
+  which means that an undefined token will not be added as a literal and
+  will simply be skipped as if the toekn is not entered.
```
[the remark this answers is not in the document]
Comment (necouchman) [reply src/configuring-guacamole.md]: Fixed via rebase.

File: src/configuring-guacamole.md (hunk 2 above)
Comment (necouchman) [reply src/configuring-guacamole.md]: Fixed via rebase.

File: src/configuring-guacamole.md [hunk 4]
```diff
@@ -2755,6 +2755,21 @@ supported:
...
+  in the connection parameter. This modifier marks a token as "optional",
+  which means that an undefined token will not be added as a literal and
+  will simply be skipped as if the token is not entered.
+
+  In some cases this will simply result in a string lacking the information
+  that would otherwise be in the token; however, in other cases this may
+  result in Guacamole generating prompts. For example, if you specify the
+  token `${GUAC_PASSWORD}` for the password field in a connection, but you
+  sign in with an SSO extension such that the user does not have a
+  `${GUAC_PASSWORD}` token, Guacamole may prompt you for the password as
```
Comment (jmuehlner) [inline src/configuring-guacamole.md:2770]: Wouldn't this be `${GUAC_PASSWORD_OPTIONAL}`?

Comment (necouchman) [reply src/configuring-guacamole.md:2770]: Actually, `${GUAC_PASSWORD:OPTIONAL}`.

**Pull request merged**